libc/test/src/unistd/readlink_test.cpp

```diff
@@ -30,8 +31,9 @@ TEST(LlvmLibcReadlinkTest, CreateAndUnlink) {
   //   3. Cleanup the symlink created in step #1.
   ASSERT_THAT(LIBC_NAMESPACE::symlink(LINK_VAL, LINK), Succeeds(0));
 
-  char buf[sizeof(LINK_VAL)];
-  ssize_t len = LIBC_NAMESPACE::readlink(LINK, buf, sizeof(buf));
+  char buf[sizeof(FILENAME)];
+  ssize_t len =
+      LIBC_NAMESPACE::readlink(LINK, buf, LIBC_NAMESPACE::strlen(FILENAME));
```

Comment:
Can you use `LIBC_NAMESPACE::internal::string_length` from https://github.com/llvm/llvm-project/blob/main/libc/src/string/string_utils.h#L90 instead.

Reply:
sure